lib/practice_exercises.rb

```diff
@@ -1,13 +1,32 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
-end
+  word_array = string.gsub(/[[:punct:]]/, "").delete(' ')
+  reversed_arrary = []
+  last_index = word_array.length
+  until 0 > last_index
+      reversed_arrary << word_array[last_index]
+      last_index -= 1
+    end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+  return word_array.downcase == reversed_arrary.join.downcase 
 end
 
+# Time Complexity: O(n2) - because there is loop inside loop
+# Space Complexity: O(1) - because no matter how many words we will loop through, prefix stay the same number of words??
+
+
+def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍